File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.java
class Solution {
    public int[] getSumAbsoluteDifferences(int[] nums) {
        int right = Arrays.stream(nums).sum();
        int left = 0, n = nums.length;
        int[] ans = new int[n];
        
        for(int i = 0; i<n; i++) {
            int val = nums[i];
            right -= val;
            ans[i] = val*i - left + right - val*(n - i - 1);
            left += val;
        }
        
        return ans;
    }
}

File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.py
class Solution(object):
    def getSumAbsoluteDifferences(self, nums):
        total = sum(nums)
        left, right = 0, total
        ans = []
        
        for i, n in enumerate(nums):
            right -= n
            ans.append(n * i - left + right - n * (len(nums) - i - 1))
            left += n
            
        return ans
        